livi: init at 0.0.6

mksafavi opened the pull request:
## Description of changes
source URL: https://gitlab.gnome.org/guidog/livi

closes #295373

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

mksafavi committed:
livi: init at 0.0.6

Aleksanaa commented on pkgs/by-name/li/livi/package.nix:
```diff
@@ -0,0 +1,48 @@
+
+  meta = with lib; {
+    homepage = "https://gitlab.gnome.org/guidog/livi";
+    description = "μPlayer A small video player targeting mobile devices";
```
```suggestion
    description = "A small video player targeting mobile devices (also named μPlayer)";
```

Aleksanaa commented on pkgs/by-name/li/livi/package.nix:
```diff
@@ -0,0 +1,48 @@
+
+stdenv.mkDerivation rec {
+  pname = "livi";
+  version = "v0.0.6";
```
```suggestion
  version = "0.0.6";
```

Aleksanaa commented on pkgs/by-name/li/livi/package.nix:
```diff
@@ -0,0 +1,48 @@
+    owner = "guidog";
+    repo = "livi";
+    domain = "gitlab.gnome.org";
+    rev = "${version}";
```
```suggestion
    rev = "v${version}";
```

Aleksanaa commented on pkgs/by-name/li/livi/package.nix:
```diff
@@ -0,0 +1,48 @@
+  buildInputs = [
+    meson
+    ninja
+    pkg-config
```
Above three should be in `nativeBuildInputs`

Aleksanaa commented on pkgs/by-name/li/livi/package.nix:
```diff
@@ -0,0 +1,48 @@
+    gst_all_1.gst-plugins-base
+    gst_all_1.gst-plugins-bad
+    gst_all_1.gst-plugins-good
+    appstream-glib
```
should be in `nativeBuildInputs`

Aleksanaa commented on pkgs/by-name/li/livi/package.nix:
```diff
@@ -0,0 +1,48 @@
+    appstream-glib
+    gtk4
+    libadwaita
+    desktop-file-utils
```
should be in `nativeBuildInputs`

Aleksanaa commented on pkgs/by-name/li/livi/package.nix:
```diff
@@ -0,0 +1,48 @@
+    homepage = "https://gitlab.gnome.org/guidog/livi";
+    description = "μPlayer A small video player targeting mobile devices";
+    license = licenses.gpl3;
+    platforms = platforms.linux;
```
```suggestion
    platforms = platforms.linux;
    mainProgram = "livi";
```

Not adding yourself to maintainers?

Aleksanaa commented:
> I have an issue with this packages. It requires gstreamer on runtime. I assumed adding gst_all_1 packages to buildInputs would fix that. But It seems like that's not the case. If I create a shell with livi, it can't find gst elements. Do I need to do anything specific to include gst in the derivation?

Oh you forgot to add `wrapGAppsHook4` to `nativeBuildInputs`. Gstreamer packages add `$GST_PLUGIN_SYSTEM_PATH_1_0` but don't create a wrapper themselves. `wrapGAppsHook4` can handle it for you. If you use other wrappers, you have to manually append it. 

> I thought it should be better if a user of the app becomes the maintainer. That being said, I'm ok with maintaining it myself too.

That's not the case. Just others will ping you for issues and build failures, and you can deal with automatic updates yourself.

mksafavi committed:
maintainers: add mksafavi

mksafavi commented:
> Oh you forgot to add `wrapGAppsHook4` to `nativeBuildInputs`. Gstreamer packages add `$GST_PLUGIN_SYSTEM_PATH_1_0` but don't create a wrapper themselves. `wrapGAppsHook4` can handle it for you. If you use other wrappers, you have to manually append it.
> 
Thanks. adding `wrapGAppsHook4` fixed the issue. 

> That's not the case. Just others will ping you for issues and build failures, and you can deal with automatic updates yourself.

I added myself to the maintainers.


Also, I squashed the commits and fixed the messages. I think it should fit the [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md) now.

Aleksanaa commented on pkgs/by-name/li/livi/package.nix:
```diff
@@ -0,0 +1,54 @@
+    gst_all_1.gstreamer
+    gst_all_1.gst-plugins-base
+    gst_all_1.gst-plugins-bad
+    gst_all_1.gst-plugins-good
```
These gst packages should be in `buildInputs`

mksafavi replied:
Oh. I miss interpreted the previous review and instead of moving the last line, I moved the whole highlighted section into `nativeBuildInput`. I'll fix this.
How do you know which packages should get into `buildInputs`?

Aleksanaa replied:
I should admit that I mostly go by feel. Packages that are used at build time should go in `nativeBuildInputs`, and packages that are linked or called at runtime should go in `buildInputs`. If you want a very accurate representation, you can add `strictDeps = true`, in this way if you don't do it correctly, you'll fail to build.

Aleksanaa commented on pkgs/by-name/li/livi/package.nix:
```diff
@@ -0,0 +1,54 @@
+    gst_all_1.gst-plugins-good
+    appstream-glib
+    gtk4
+    libadwaita
```
`gtk4` and `libadwaita` should also be in `buildInputs`

Aleksanaa commented on pkgs/by-name/li/livi/package.nix:
```diff
@@ -0,0 +1,54 @@
+  meta = with lib; {
+    homepage = "https://gitlab.gnome.org/guidog/livi";
+    description = "A small video player targeting mobile devices (also named μPlayer)";
+    license = licenses.gpl3;
```
```suggestion
    license = licenses.gpl3Plus;
```

mksafavi replied:
> 

I think it should be changed into `gpl3Only` based on this https://gitlab.gnome.org/guidog/livi/-/blob/main/COPYING
I seems like `gpl3` is deprecated but I didn't get any warnings in the build. 
Do I know why my builds don't care about the meta section? It was the same with the maintainer field as well. I didn't notice any issues with it but Borg caught it and failed.

Aleksanaa replied:
> I think it should be changed into gpl3Only based on this https://gitlab.gnome.org/guidog/livi/-/blob/main/COPYING

It's at the top of c source files. Also the `flathub` uses `gpl3Plus`.

> It seems like gpl3 is deprecated but I didn't get any warnings in the build.
Do you know why my builds don't care about the meta section? It was the same with the maintainer field as well. I didn't notice any issues with it but Borg caught it and failed.

Because It does not use these sections.

Aleksanaa reviewed (approved):
lgtm. Please squash your commits into `livi: init at 0.0.6` (except adding maintainer)

mksafavi commented on pkgs/by-name/li/livi/package.nix:
```diff
@@ -0,0 +1,55 @@
+{ lib
```
Should I order the args alphabetically here?
I noticed that some packages put builtins like lib, stdenv,... first and then order the deps alphabetically.

Aleksanaa replied:
Not needed. You can also sort them in the order they are used. It's up to you. But `lib` should be placed in first order.
